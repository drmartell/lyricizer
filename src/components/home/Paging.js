import React from 'react';
import PropTypes from 'prop-types';

const Paging = ({ onIncrement, onDecrement, page, count }) => {
  const maxPage = Math.max(Math.floor(count / 25), 1);
  return (
    <>
      <button onClick={onDecrement}>&larr;</button>
      <span>{page} / {maxPage}</span>
      <button onClick={onIncrement}>&rarr;</button>
    </>
  );};

Paging.propTypes = {
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
  page: PropTypes.number.isRequired,
  count: PropTypes.number.isRequired
};

export default Paging;
