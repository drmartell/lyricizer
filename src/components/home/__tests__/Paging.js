import React from 'react';
import { shallow } from 'enzyme';
import Paging from '../Paging';

describe('Paging Component', () => {
  it('should render Paging', () => {
    const wrapper = shallow(<Paging />);
    expect(wrapper).toMatchSnapshot();
  });
});
