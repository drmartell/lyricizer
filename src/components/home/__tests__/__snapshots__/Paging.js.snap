// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging Component should render Paging 1`] = `
<Fragment>
  <button>
    ←
  </button>
  <span>
     / 
    NaN
  </span>
  <button>
    →
  </button>
</Fragment>
`;
